### PR TITLE
Tracking system, editing collection name

### PR DESCRIPTION
**Pull request by simotae14**
What does it do?
Added the event didEditFieldOnContentTypeInCTB when we click Finish in the Edit modal after an update of the every field name

Why is it needed?
As a Strapi PM, the PM wants the didEditFieldOnContentType event to be available in the tracking system, so that analytics on the effective edits of a content type’s field can be created.

How to test it?
 - As a Strapi PM
   - when I access the content type builder section from the side menu
     - then I want the didEditFieldOnContentTypeInCTB to be fired 
   - when I create a graph on Amplitude
     - then I want the didEditFieldOnContentTypeInCTB to be available and working

### CODE REVIEW

File: packages/core/content-type-builder/admin/src/components/FormModal/index.js [hunk 1]
```diff
@@ -510,8 +510,13 @@ const FormModal = () => {
         } else {
           // We cannot switch from collection type to single when the modal is making relations other than oneWay or manyWay
           if (canEditContentType(allDataSchema, modifiedData)) {
+            // allDataSchema.contentType.schema.displayName
```
Comment (ronronscelestes) [inline packages/core/content-type-builder/admin/src/components/FormModal/index.js:513]: is this a leftover? 👀

File: packages/core/content-type-builder/admin/src/components/FormModal/index.js [hunk 2]
```diff
@@ -510,8 +510,13 @@ const FormModal = () => {
...
             onCloseModal();
 
+            if (modifiedData?.displayName !== allDataSchema?.contentType?.schema?.displayName) {
+              trackUsage('didEditFieldOnContentTypeInCTB');
```
Comment (ronronscelestes) [inline packages/core/content-type-builder/admin/src/components/FormModal/index.js:517]: just a personal opinion: `didEditFieldOnContentTypeInCTB` doesn't seem clear that it's only the display name of the content type that was changed, it feels like any field of the content type could have been changed, wdyt?

Comment (simotae14) [reply packages/core/content-type-builder/admin/src/components/FormModal/index.js:517]: yes I think the same, I have asked the same directly to Marco and now in the jira ticket, thanks @ronronscelestes

File: packages/core/content-type-builder/admin/src/components/FormModal/index.js (hunk 1 above)
Comment (simotae14) [reply packages/core/content-type-builder/admin/src/components/FormModal/index.js:513]: removed

File: packages/core/content-type-builder/admin/src/components/FormModal/index.js (hunk 2 above)
Comment (simotae14) [reply packages/core/content-type-builder/admin/src/components/FormModal/index.js]: we have changed the event to `didEditFieldNameOnContentType `

Comment (ronronscelestes) [reply packages/core/content-type-builder/admin/src/components/FormModal/index.js]: sounds great :)

**Review by ronronscelestes — APPROVED** [review]
LGTM 💃

**Comment by simotae14** [comment]
I have refactored the code based on Marco's tests. We want to trigger the event every time we modify the field name

File: packages/core/content-type-builder/admin/src/components/FormModal/index.js [hunk 3]
```diff
@@ -942,6 +942,13 @@ const FormModal = () => {
 
   const schemaKind = get(contentTypes, [targetUid, 'schema', 'kind']);
 
+  const isEditingFieldName =
```
Comment (joshuaellis) [inline packages/core/content-type-builder/admin/src/components/FormModal/index.js:945]: This calc will run on every render when it's not needed, why not always pass the event handler and run that check in said event handler?

**Review by joshuaellis — CHANGES_REQUESTED** [review]

File: packages/core/content-type-builder/admin/src/components/FormModal/index.js (hunk 3 above)
Comment (simotae14) [reply packages/core/content-type-builder/admin/src/components/FormModal/index.js]: Ok, thanks for the suggestion, modified

**Review by joshuaellis — APPROVED** [review]

**Review by ronronscelestes — APPROVED** [review]
LGTM 💃

**Pull request merged**